docs/data-sources/deployment_templates.md

```diff
@@ -0,0 +1,315 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "ec_deployment_templates Data Source - terraform-provider-ec"
+subcategory: ""
+description: |-
+  Use this data source to retrieve a list of available deployment templates.
+---
+
+# ec_deployment_templates (Data Source)
+
+Use this data source to retrieve a list of available deployment templates.
+
+## Example Usage
+
+```terraform
+data "ec_deployment_templates" "example" {
+  region = "us-east-1"
+}
+```
+
+### Creating a deployment from a deployment template
+
+```terraform
+data "ec_deployment_templates" "all_templates" {
+  region = "aws-eu-west-1"
+}
+
+resource "ec_deployment" "my_deployment" {
+  name                   = "My Deployment"
+  version                = "8.12.2"
+  region                 = data.ec_deployment_templates.all_templates.region
+  deployment_template_id = data.ec_deployment_templates.all_templates.templates.0.id
+
+  elasticsearch = data.ec_deployment_templates.all_templates.templates.0.elasticsearch
```

Comment:
I'm curious how `node_roles` and `node_types` are handled in this case? IIRC, templates define both `node_types` and `node_roles` but I guess we cannot use both in deployment payload.

Reply:
Never mind :)
The datasource doesn't return either `node_roles` or `node_types`.

Reply:
It works because the template currently doesn't contain either field.

But actually I might remove this example since it is really not that useful. Since the code anyway uses the template as the default and when using this pattern you can't override any values, which pretty much everyone will want to do.